src/playground-code-editor.ts

```diff
@@ -271,7 +296,28 @@ export class PlaygroundCodeEditor extends LitElement {
       >;
       for (const prop of changedTyped.keys()) {
         switch (prop) {
+          case 'documentKey': {
+            const docKey = this.documentKey ?? this._fallbackDocInstance;
```

Comment:
Curious about that sequence I mentioned in the other comment. If we go (1) undefined (2) defined (3) undefined -- then does it make sense for there to be common history between (1) and (3)?

I think if we did not want it to be shared, the change could maybe just be this?:

```suggestion
            const docKey = this.documentKey ?? {};
```


I'm actually totally not sure what the behavior should be, curious what you think.

Reply:
You are a genius. Your test case is great. I've codified it and played with both behaviors.

The behavior that "feels" right is what you suggested, that unsetting the `documentKey` should clear history. And I was mistakenly thinking that this was a breaking change. But it isn't. If the user never uses documentKey they get a single history.

Added a new test in newest commit for this case.